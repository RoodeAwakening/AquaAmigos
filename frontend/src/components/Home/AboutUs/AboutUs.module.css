.about_us__container {

  
}

.about_us__titles {
  margin-bottom: 4em;
  text-align: center;
}

.about_us__members{
  height: auto;
  width: 90vw;
background-color: gold;
position: relative;
}

.about_us__members_alignment{
  display: flex;
  justify-content: center;
  align-items: center;
  /* wrap content to another row if overflow */
  flex-wrap: wrap;
  position: absolute;
}


.about_us__titles h1 {
  font-size: 3em;
  margin-bottom: 0.8em;
  font-weight: 100;
  text-transform: capitalize;
  letter-spacing: 2px;
}
.about_us__titles h4 {
  font-size: 0.9em;
  font-weight: 100;
  color: dimgray;
}
.about_us__frame {
  position: relative;
  width: 200px;
  height: 255px;
  margin: 3em;
  overflow: hidden;
  /* center content */
  display: flex;
  justify-content: center;
  box-shadow: #615d5da1 0px 3px 10px;
}

.about_us__frame img {
  position: absolute;
  height: 100%;
  width: auto;
  /* center */
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.about_us__frame:hover > .about_us__in_out_alert {
  transform: translateY(0);
  
}
.about_us__in_out_alert {
  position: absolute;
  display: flex;
  padding: 2em 0;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
  width: 100%;
  left: 0;
  bottom: 0;
  background: white;
  transition: all 0.5s;
  transform: translateY(100%);
}

.about_us__in_out_alert > h3 {
  margin: 0;
  line-height: 1.3;
}
.about_us__in_out_alert > h2:first-of-type {
  font-size: 1em;
  font-weight: 400;
  letter-spacing: 1.6px;
  color: #545252;
}
.about_us__in_out_alert > h3:last-child {
  font-size: 1em;
  font-weight: 400;
  letter-spacing: 1.6px;
  color: #545252;
}
.userName{
  text-align: center;
  width: 100%;
  background-color: rgba(9, 9, 245, 0.459);
}

.userTitle, .aboutMe{
  text-align: center;
}
