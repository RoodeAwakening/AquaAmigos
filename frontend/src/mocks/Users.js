export default [
  {
    id: 5,
    userName: 'MrMember',
    userIcon: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSRMlcYYdpn9HleCLqzIcI8BswbjHc6oSjz6VbL8IMa5Br-IGDklWZSaY3J5N-HFgeZSms&usqp=CAU',
    email: 'member@aqua.com',
    password: 'asdf',
    role: 'member',
    position: 0,
    title: 'member',
    first_name: 'Chris',
    last_name: 'Jacobs',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'New to the club, I am looking for a place to start.',
  },
  {
    id: 1,
    userName: 'MrPresident',
    userIcon: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTcg4Y51XjQ-zSf87X4nUPTQzsF83eFdZswTg&usqp=CAU',
    email: 'president@aqua.com',
    password: 'admin',
    role: 'admin',
    position: 1,
    title: 'President',
    first_name: 'Jacob',
    last_name: 'Smith',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'Loved diving since the age of 5 and now I get to help run a scuba club.',
  },
  {
    id: 2,
    userName: 'VicePresident',
    userIcon: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQNXtFTglXjnFYetM_ceM1h8Uzx6-FS19WHObUPV2UUYZdOHGNgrDBppJ16sRAJQZ-DdoI&usqp=CAU',
    email: 'vp@aqua.com',
    password: 'asdf',
    role: 'member',
    position: 2,
    title: 'Vice President',
    first_name: 'Susan',
    last_name: 'Williams',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'Look how long I can hold my breath!',
  },
  {
    id: 4,
    userName: 'MrsTreasurer',
    userIcon: 'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8dXNlciUyMHByb2ZpbGV8ZW58MHx8MHx8&w=1000&q=80',
    email: 'treasurer@aqua.com',
    password: 'asdf',
    role: 'editor',
    position: 4,
    title: 'Treasurer',
    first_name: 'Jane',
    last_name: 'Doe',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'Keeping the club afloat is my passion.',
  },
  {
    id: 3,
    userName: 'MrSecretary',
    userIcon: 'https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50?s=200',
    email: 'secretary@aqua.com',
    password: 'asdf',
    role: 'editor',
    position: 3,
    title: 'Secretary',
    first_name: 'John',
    last_name: 'Doe',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'I love to help people with their diving needs.',
  },
  {
    id: 6,
    userName: 'TacosForMe',
    userIcon: 'https://images.squarespace-cdn.com/content/v1/53ed0e3ce4b0c296acaeae80/1584577511464-8FDZYWQVXUI1OBS4VTZP/Bonneville14082-Edit-DHWEB%2BNick%2BFerguson%2BDenver%2BBroncos%2BHeadshot%2BPhotography%2Bby%2BAaron%2BLucy%2BDenver%2BColorado%2BHeadshots%2BPhotographer.jpg?format=2500w',
    email: 'iLike2Dive@aol.com',
    password: 'avbi',
    role: 'editor',
    position: 5,
    title: 'Trustee',
    first_name: 'Greg',
    last_name: 'Williams',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'Making sure the club is running smoothly is my passion. When I am not diving I like to cook and bake.',
  },
  {
    id: 8,
    userName: 'DiveClub',
    userIcon: 'https://images.unsplash.com/photo-1518791841217-8f162f1e1131?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60',
    email: 'lvai4@aol.com',
    password: 'avb24i',
    role: 'editor',
    position: 6,
    title: 'Web Committee',
    first_name: 'Logan',
    last_name: 'Vai',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'I love to help the club grow and make sure it is a great place to dive.',
  },
  {
    id: 7,
    userName: 'Tomorrow',
    userIcon: 'https://ephotovn.com/wp-content/uploads/2019/10/basic-headshot-retouching-af.jpg',
    email: 'iDive@aol.com',
    password: 'avb24i',
    role: 'editor',
    position: 5,
    title: 'Trustee',
    first_name: 'Brian',
    last_name: 'DeSimone',
    created_at: '2020-05-06T14:52:37.000Z',
    updated_at: '2020-05-06T14:52:37.000Z',
    aboutMe: 'My love for the ocean is what makes me a great swimmer.',
  },

];

// President - position: 1 - role: admin
// Vide-President - position: 2 - role: admin
// Secretary - position: 3 - role: editor
// Treasurer - position: 4 - role: editor
// Trustee - position: 5 - role: editor
// committee chairman - position: 6 - role: editor
// member - position: 0 - role: member
