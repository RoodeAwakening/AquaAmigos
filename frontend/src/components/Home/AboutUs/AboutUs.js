import React from 'react';
import i18next from 'i18next';
import styles from './AboutUs.module.css';

import Users from '../../../mocks/Users';
import verifyTeamMember from './helper/sortTeamMembers';

function AboutUs() {
  const leaders = verifyTeamMember(Users).map((user) => {
    return (

      <div className={styles.about_us__frame}>
        <img
          src={user.userIcon}
          alt="Team Members"
        />
        <div className={styles.about_us__in_out_alert}>
          <h3 className={styles.userTitle}>{user.title}</h3>
          <h3 className={styles.userName}>{`${user.first_name} ${user.last_name}`}</h3>
          <h3 className={styles.aboutMe}>{user.aboutMe}</h3>
        </div>
      </div>
    );
  });

  return (
    <section className={styles.about_us__container}>
      <div className={styles.about_us__titles}>
        <h1>{i18next.t('msg_team_members')}</h1>
        <h4>{i18next.t('msg_team_members_sub')}</h4>
      </div>
      <div className={styles.about_us__members}>
        <div className={styles.about_us__members_alignment}>
          {leaders}
        </div>
      </div>
    </section>
  );
}

export default AboutUs;
