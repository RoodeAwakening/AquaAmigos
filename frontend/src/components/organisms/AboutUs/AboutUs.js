import React from 'react';
import styles from './AboutUs.module.css';

import Users from '../../../mocks/Users';
import verifyTeamMember from '../../../helpers/sortTeamMembers';
import SectionTitle from '../../molecules/SectionTitle/SectionTitle';
import ResponsiveBlock from '../../atoms/ResponsiveBlock/ResponsiveBlock';
import ContentBlock from '../../molecules/ContentBlock/ContentBlock';

function AboutUs() {
  const leaders = verifyTeamMember(Users).map((user) => {
    return (
      <>
        <ContentBlock
          key={user.id}
          title={`${user.title} - ${user.first_name} ${user.last_name}`}
          description={`${user.aboutMe} `}
          articleImage={user.userIcon}
        />
      </>
    );
  });

  return (
    <div className={styles.about_us__container}>
      <SectionTitle header="msg_team_members" subHeader="msg_team_members_sub" />
      <ResponsiveBlock>
        {leaders}
      </ResponsiveBlock>
    </div>
  );
}

export default AboutUs;
