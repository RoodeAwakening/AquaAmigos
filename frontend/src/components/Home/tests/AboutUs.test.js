// test to see if the AboutUs component is rendering correctly
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import AboutUs from '../AboutUs/AboutUs';
import Users from '../../../mocks/Users';

configure({ adapter: new Adapter() });
global.confirm = () => true;

describe('AboutUs Component', () => {
  let aboutUs;

  beforeEach(() => {
    aboutUs = shallow(<AboutUs />);
  });

  afterEach(() => {
    aboutUs.unmount();
  });

  it('Should render the AboutUs component to match the snapshot', () => {
    expect(toJson(aboutUs)).toMatchSnapshot();
  });

  it('Should render all Users that dont have a position number of 0', () => {
    const members = Users.filter(user => user.position > 0);
    // sort members from lowest to highest position number
    members.sort((a, b) => a.position - b.position);

    members.map((user) => {
      return (
        <div className="about_us__frame">
          <img
            src={user.userIcon}
            alt="Team Members"
          />
          <div className="about_us__in_out_alert">
            <h3 className="userTitle">{user.title}</h3>
            <h3 className="userName">{`${user.first_name} ${user.last_name}`}</h3>
            <h3 className="aboutMe">{user.aboutMe}</h3>
          </div>
        </div>
      );
    });

    expect(aboutUs.find('.about_us__members').children().length).toEqual(1);
  });
});
