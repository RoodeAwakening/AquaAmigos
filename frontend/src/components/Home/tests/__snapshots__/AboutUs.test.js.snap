// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs Component Should render the AboutUs component to match the snapshot 1`] = `
<section
  className="about_us__container"
>
  <div
    className="about_us__titles"
  >
    <h1 />
    <h4 />
  </div>
  <div
    className="about_us__members"
  >
    <div
      className="about_us__members_alignment"
    >
      <div
        className="about_us__frame"
      >
        <img
          alt="Team Members"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTcg4Y51XjQ-zSf87X4nUPTQzsF83eFdZswTg&usqp=CAU"
        />
        <div
          className="about_us__in_out_alert"
        >
          <h3
            className="userTitle"
          >
            President
          </h3>
          <h3
            className="userName"
          >
            Jacob Smith
          </h3>
          <h3
            className="aboutMe"
          >
            Loved diving since the age of 5 and now I get to help run a scuba club.
          </h3>
        </div>
      </div>
      <div
        className="about_us__frame"
      >
        <img
          alt="Team Members"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQNXtFTglXjnFYetM_ceM1h8Uzx6-FS19WHObUPV2UUYZdOHGNgrDBppJ16sRAJQZ-DdoI&usqp=CAU"
        />
        <div
          className="about_us__in_out_alert"
        >
          <h3
            className="userTitle"
          >
            Vice President
          </h3>
          <h3
            className="userName"
          >
            Susan Williams
          </h3>
          <h3
            className="aboutMe"
          >
            Look how long I can hold my breath!
          </h3>
        </div>
      </div>
      <div
        className="about_us__frame"
      >
        <img
          alt="Team Members"
          src="https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50?s=200"
        />
        <div
          className="about_us__in_out_alert"
        >
          <h3
            className="userTitle"
          >
            Secretary
          </h3>
          <h3
            className="userName"
          >
            John Doe
          </h3>
          <h3
            className="aboutMe"
          >
            I love to help people with their diving needs.
          </h3>
        </div>
      </div>
      <div
        className="about_us__frame"
      >
        <img
          alt="Team Members"
          src="https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8dXNlciUyMHByb2ZpbGV8ZW58MHx8MHx8&w=1000&q=80"
        />
        <div
          className="about_us__in_out_alert"
        >
          <h3
            className="userTitle"
          >
            Treasurer
          </h3>
          <h3
            className="userName"
          >
            Jane Doe
          </h3>
          <h3
            className="aboutMe"
          >
            Keeping the club afloat is my passion.
          </h3>
        </div>
      </div>
      <div
        className="about_us__frame"
      >
        <img
          alt="Team Members"
          src="https://images.squarespace-cdn.com/content/v1/53ed0e3ce4b0c296acaeae80/1584577511464-8FDZYWQVXUI1OBS4VTZP/Bonneville14082-Edit-DHWEB%2BNick%2BFerguson%2BDenver%2BBroncos%2BHeadshot%2BPhotography%2Bby%2BAaron%2BLucy%2BDenver%2BColorado%2BHeadshots%2BPhotographer.jpg?format=2500w"
        />
        <div
          className="about_us__in_out_alert"
        >
          <h3
            className="userTitle"
          >
            Trustee
          </h3>
          <h3
            className="userName"
          >
            Greg Williams
          </h3>
          <h3
            className="aboutMe"
          >
            Making sure the club is running smoothly is my passion. When I am not diving I like to cook and bake.
          </h3>
        </div>
      </div>
      <div
        className="about_us__frame"
      >
        <img
          alt="Team Members"
          src="https://ephotovn.com/wp-content/uploads/2019/10/basic-headshot-retouching-af.jpg"
        />
        <div
          className="about_us__in_out_alert"
        >
          <h3
            className="userTitle"
          >
            Trustee
          </h3>
          <h3
            className="userName"
          >
            Brian DeSimone
          </h3>
          <h3
            className="aboutMe"
          >
            My love for the ocean is what makes me a great swimmer.
          </h3>
        </div>
      </div>
      <div
        className="about_us__frame"
      >
        <img
          alt="Team Members"
          src="https://images.unsplash.com/photo-1518791841217-8f162f1e1131?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
        />
        <div
          className="about_us__in_out_alert"
        >
          <h3
            className="userTitle"
          >
            Web Committee
          </h3>
          <h3
            className="userName"
          >
            Logan Vai
          </h3>
          <h3
            className="aboutMe"
          >
            I love to help the club grow and make sure it is a great place to dive.
          </h3>
        </div>
      </div>
    </div>
  </div>
</section>
`;
